store/postgres/migrations/2024-01-05-170000_ds_corruption_fix_up/up.sql

```diff
@@ -0,0 +1,96 @@
+/*
+
+
+This migration exists to fix up DB inconsistencies caused by a bug that was already fixed in this PR:
+https://github.com/graphprotocol/graph-node/pull/5083
+
+What it does is:
+1. select latest_ethereum_block_number from subgraphs.subgraph_deployment where deployment = '$Qm..';
+2. With that value, check if there are any data sources with a higher block:
+   select count(*) from sgd*.data_sources$ where lower(block_range) > $latest_block;
+3. If there are, then we need to intervene in that subgraph by removing all such data sources:
+   delete from sgd*.data_sources$ where lower(block_range) > $latest_block;
+4. It also unclamps any data sources that need unclamping (even though I don't think we ever clamp them).
+*/
+
+CREATE TEMPORARY TABLE temp_sgd_last_block (
+    deployment_schema text,
+    last_block_from_registry numeric,
+    is_ok boolean,
+    is_upper_limit_too_high boolean
+);
+
+-- collect the latest block number from SG deployment details
+INSERT INTO temp_sgd_last_block (deployment_schema, last_block_from_registry)
+SELECT d.name, latest_ethereum_block_number
+  FROM subgraphs.subgraph_deployment AS sd
+  JOIN deployment_schemas AS d ON sd.deployment = d.subgraph
+ WHERE d.name IN (SELECT relnamespace::regnamespace::name
+                    FROM pg_class
+                   WHERE relname LIKE 'data_sources$%' AND relkind = 'r'
+                 );
```

Comment:
I don't quite follow what's going on here - this is to find all deployments that have a `data_sources$` table? Might be simpler to join on `information_schema.tables.table_schema where information_schema.tables.table_name = 'data_sources$'`

Reply:
Exactly.  TBH `information_schema` is something I never use, as it's limited in most places, and otherwise I'm way more familiar with the catalog tables and views.